xbmc/filesystem/File.cpp

```diff
@@ -283,7 +283,9 @@ bool CFile::Open(const CURL& file, const unsigned int flags)
         return m_pFile->Open(url);
```

Comment:
I wouldn't be sure here.... if cache is cleared between the check and the call it could be null.